armory/paths.py

```diff
@@ -5,40 +5,53 @@
 import json
 import logging
 import os
-from pathlib import Path
-
 
 logger = logging.getLogger(__name__)
 
 
-PROJECT_ROOT = Path(__file__).parents[1]
-CWD = os.getcwd()
-USER = os.path.expanduser("~")
-USER_ARMORY = os.path.join(USER, ".armory")
-USER_KERAS = os.path.join(USER, ".keras")
-ARMORY_CONFIG = os.path.join(USER_ARMORY, "config.json")
-if os.path.exists(ARMORY_CONFIG):
-    try:
-        with open(ARMORY_CONFIG) as f:
-            config = json.load(f)
-    except json.decoder.JSONDecodeError:
-        logger.exception(f"Armory config file {ARMORY_CONFIG} could not be decoded")
-        raise
-    except OSError:
-        logger.exception(f"Armory config file {ARMORY_CONFIG} could not be read")
-        raise
-else:
-    os.makedirs(USER_ARMORY, exist_ok=True)
-    config = {"cached_dataset_dir": os.path.join(USER_ARMORY, "datasets")}
-    with open(ARMORY_CONFIG, "w") as f:
-        json.dump(config, f)
-
-DATASETS = config.get("cached_dataset_dir") or os.path.join(USER_ARMORY, "datasets")
-EXTERNAL_REPOS = os.path.join(DATASETS, "external_repos")
-MODELS = config.get("models") or os.path.join(USER_ARMORY, "models")
-TMP = config.get("tmp") or os.path.join(USER_ARMORY, "tmp")
-OUTPUTS = config.get("outputs") or os.path.join(USER_ARMORY, "outputs")
-
-DOCKER = "/root"
-DOCKER_ARMORY = os.path.join(DOCKER, ".armory")
-DOCKER_TMP = Path(os.path.join(DOCKER_ARMORY, "tmp")).as_posix()
+def _parse_global_config(path, armory_dir):
+    if os.path.exists(path):
+        try:
+            with open(path) as f:
+                config = json.load(f)
+        except json.decoder.JSONDecodeError:
+            logger.exception(f"Armory config file {path} could not be decoded")
+            raise
+        except OSError:
+            logger.exception(f"Armory config file {path} could not be read")
+            raise
+    else:
```

Comment:
If the config file exists but there are issues parsing it (e.g., it's an old config like in my case, or because they manually modified it), we should try to raise a more understandable error than:
```
(armory-base) davidslater@aleph-5:~/armory$ python -m armory run examples/fgm_attack.json 
Traceback (most recent call last):
  File "/Users/davidslater/opt/anaconda3/envs/armory-base/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/Users/davidslater/opt/anaconda3/envs/armory-base/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/Users/davidslater/git/armory/armory/__main__.py", line 204, in <module>
    main()
  File "/Users/davidslater/git/armory/armory/__main__.py", line 200, in main
    func(sys.argv[2:], prog, description)
  File "/Users/davidslater/git/armory/armory/__main__.py", line 102, in run
    rig = Evaluator(args.filepath)
  File "/Users/davidslater/git/armory/armory/eval/evaluator.py", line 27, in __init__
    self.host_paths = HostPaths()
  File "/Users/davidslater/git/armory/armory/paths.py", line 46, in __init__
    config = _parse_global_config(self.armory_config, self.armory_dir)
  File "/Users/davidslater/git/armory/armory/paths.py", line 32, in _parse_global_config
    os.makedirs(config["cached_dataset_dir"], exist_ok=True)
KeyError: 'cached_dataset_dir'
```

Perhaps that's something to add with the `armory configure` issue?

Reply:
Thanks. Added a temporary error which we can modify when #165 is complete